### PR TITLE
Add @tangleid/mnid package

### PR DESCRIPTION
**Pull request by ender503**
To make the package can be easily maintained, move the MNID functions
to "@tangleid/mnid" the package, and convert it to TypeScript project.

Since TypeScript compiler does not clean up the output directory
before compiling, add script to up output directory and run it before
building the packages.

Since the README documentation cannot be generated from the TypeScript
files directly, compile these files to the JavaScript, then generate
documentation from them.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,4 +4,5 @@ node_js:
 install:
   - yarn run init
 script:
-  - yarn run test
+  - yarn run lint
```
Comment (jserv) [inline .travis.yml]: `lint` should be included in another pull request.

File: packages/mnid/LICENSE [hunk 2]
```diff
@@ -0,0 +1,98 @@
+The MIT License (MIT)
+
+Copyright (C) 2018 BiiLabs, Co. Ltd. and Contributors
```
Comment (jserv) [inline packages/mnid/LICENSE]: Should be `2018-2019`

Comment (jserv) [reply packages/mnid/LICENSE]: The formal name is `BiiLabs Co., Ltd.` Be aware of the comma.

**Comment by jserv** [comment]
Top-level `README.md` should be updated accordingly.

File: .travis.yml (hunk 1 above)
Comment (ender503) [reply .travis.yml]: I'll send `lint` in the coming pull request. 👍

File: packages/mnid/LICENSE (hunk 2 above)
Comment (ender503) [reply packages/mnid/LICENSE]: I notice that this already fixed in the `develop` branch, so I rebased this branch.

**Comment by ender503** [comment]
I updated the top-level `README.md`. 👌

File: .travis.yml (hunk 1 above)
Comment (ender503) [reply .travis.yml]: The `lint` pull request is blocked by this pull request.
